bundles/nl.asml.matala.product/src/nl/asml/matala/product/generator/ValidationHelper.java

```diff
@@ -0,0 +1,70 @@
+package nl.asml.matala.product.generator;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.function.Function;
+
+
+
+import nl.esi.comma.actions.actions.Action;
+import nl.esi.comma.actions.actions.AssignmentAction;
+import nl.esi.comma.actions.actions.ForAction;
+import nl.esi.comma.actions.actions.IfAction;
+import nl.esi.comma.actions.actions.RecordFieldAssignmentAction;
+import nl.esi.comma.expressions.expression.Expression;
+import nl.esi.comma.expressions.expression.ExpressionRecordAccess;
+import nl.esi.comma.expressions.generator.ExpressionsCommaGenerator;
+
+public class ValidationHelper {
+	
+	/**
+	 * Get variables based on the action type
+	 * */
+	
+	public static Map<String, Object> action(Action action) {
+		if (action instanceof AssignmentAction) {
+			AssignmentAction a = (AssignmentAction) action;
+		    return assignmentAction(a);			
+		} 
+		else if (action instanceof RecordFieldAssignmentAction) {
+			RecordFieldAssignmentAction a = (RecordFieldAssignmentAction) action;
+			ExpressionRecordAccess access = (ExpressionRecordAccess) a.getFieldAccess();
+			String record = (new ExpressionsCommaGenerator()).exprToComMASyntax(access.getRecord()).toString();			
+			Map<String, Object> map = new HashMap<>();
```

Comment:
Instead of using a Map with magic string as keys, please us a tuple (i.e. Xtend data object). This makes the API much more intuitive and more robust.